### PR TITLE
[DLP] Implemented a sample for Inspect String with Augmented InfoTypes

### PR DESCRIPTION
**Pull request by bhavika-dhanwani**
Implemented a sample for [augmenting built-in infoType detectors by including a built-in detector](https://cloud.google.com/dlp/docs/creating-custom-infotypes-dictionary#augment_a_built-in_infotype_detector)

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [X] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: dlp/snippets/src/test/java/dlp/snippets/InspectTests.java [hunk 1]
```diff
@@ -407,4 +407,13 @@ public void testInspectWithHotwordRules() throws Exception {
     assertThat(output).contains("Findings: 2");
     assertThat(output).contains("Info type: C_MRN");
   }
+
+  @Test
+  public void testInspectStringAugmentInfoType() throws Exception {
+    InspectStringAugmentInfoType.inspectStringAugmentInfoType(
+        "bdp-2059-is-31084", "patient: quasimodo");
```
Comment (annadata) [inline dlp/snippets/src/test/java/dlp/snippets/InspectTests.java:414]: Do not hardcode the projectId. Use PROJECT_ID like the other tests.

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+              .setType(ByteContentItem.BytesType.TEXT_UTF8)
+              .setData(ByteString.copyFromUtf8(textToInspect))
+              .build();
+      ContentItem item = ContentItem.newBuilder().setByteItem(byteItem).build();
+
+      // Construct the custom word list to be detected.
+      CustomInfoType.Dictionary wordList =
+          CustomInfoType.Dictionary.newBuilder()
+              .setWordList(
+                  CustomInfoType.Dictionary.WordList.newBuilder().addWords("quasimodo").build())
```
Comment (annadata) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:65]: Please parameterise the words and send it from caller.

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+public class InspectStringAugmentInfoType {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // The string to de-identify.
+    String textToInspect = "patient: quasimodo";
```
Comment (annadata) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:42]: How about "The patients name is Quasimodo"

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+      // Construct the custom word list to be detected.
+      CustomInfoType.Dictionary wordList =
+          CustomInfoType.Dictionary.newBuilder()
+              .setWordList(
+                  CustomInfoType.Dictionary.WordList.newBuilder().addWords("quasimodo").build())
+              .build();
+
+      InfoType infoType = InfoType.newBuilder().setName("PERSON_NAME").build();
+      // Construct the custom infotype detector associated with the word list.
```
Comment (annadata) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:69]: // Construct a custom infotype detector by augmenting the PERSON_NAME detector with a word list

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import com.google.privacy.dlp.v2.Finding;
+import com.google.privacy.dlp.v2.InfoType;
+import com.google.privacy.dlp.v2.InspectConfig;
+import com.google.privacy.dlp.v2.InspectContentRequest;
+import com.google.privacy.dlp.v2.InspectContentResponse;
+import com.google.privacy.dlp.v2.LocationName;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+
+// [START dlp_inspect_augment_infotypes]
```
Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:32]: Move this region tag to line 18, before the import statements.

Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:32]: Move this region tag to line 18, before the import statements.

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // The string to de-identify.
+    String textToInspect = "The patient's name is quasimodo";
+    // The string to be additionally matched
+    String textToAugment = "quasimodo";
+    inspectStringAugmentInfoType(projectId, textToInspect, textToAugment);
+  }
+
+  // Inspects the provided text.
```
Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:47]: Can the description be a bit more elaborate, mentioning the use of a custom word to detect.

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+public class InspectStringAugmentInfoType {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // The string to de-identify.
+    String textToInspect = "The patient's name is quasimodo";
+    // The string to be additionally matched
```
Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:42]:
```suggestion
    // The string to be additionally matched.
```

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+      // Construct the custom word list to be detected.
+      CustomInfoType.Dictionary wordList =
+          CustomInfoType.Dictionary.newBuilder()
+              .setWordList(
+                  CustomInfoType.Dictionary.WordList.newBuilder().addWords(textToAugment).build())
+              .build();
+
+      InfoType infoType = InfoType.newBuilder().setName("PERSON_NAME").build();
+      //  Construct a custom infotype detector by augmenting the PERSON_NAME detector with a word
```
Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java:70]:
```suggestion
      // Construct a custom infotype detector by augmenting the PERSON_NAME detector with a word
```

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java (hunk 8 above)
Comment (bhavika-dhanwani) [reply dlp/snippets/src/main/java/dlp/snippets/InspectStringAugmentInfoType.java]: Resolved the suggested changes

**Review by Sita04 — APPROVED** [review]

**Pull request merged**